packages/hardhat-core/src/internal/util/scripts-runner.ts

```diff
@@ -33,7 +33,7 @@ export async function runScript(
     });
 
     childProcess.once("close", (status) => {
-      log(`Script ${scriptPath} exited with status code ${status}`);
+      log(`Script ${scriptPath} exited with status code ${status ?? 0}`);
```

Comment:
Weird, the [docs](https://nodejs.org/api/child_process.html#event-close) say that this is always a number. I wonder if this is a bug in `@types/node`.